### PR TITLE
Update react native to 0.30 and some other cleanup

### PR DESCRIPTION
**Pull request by chrisnojima**

### CODE REVIEW

File: react-native/make-shim.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+const fs = require('fs')
```
Comment (chrisnojima) [inline react-native/make-shim.js]: instead of having this all in the package.json lets just call this

File: react-native/package.json [hunk 2]
```diff
@@ -9,29 +9,29 @@
...
+    "wrap": "npm prune && npm shrinkwrap --dev",
+    "postinstall": "npm run sync; node make-shim net ; node make-shim tls ; node make-shim msgpack"
   },
   "dependencies": {
-    "buffer": "4.5.1",
-    "deep-diff": "^0.3.4",
-    "framed-msgpack-rpc": "git://github.com/keybase/node-framed-msgpack-rpc#3da13a39e63aaa97a36b0e88fada60b974393a20",
+    "buffer": "4.7.1",
+    "deep-diff": "0.3.4",
+    "framed-msgpack-rpc": "git://github.com/keybase/node-framed-msgpack-rpc#e3c96a2ebb8350836f5bae9710b89d04fb1fc87a",
```
Comment (chrisnojima) [inline react-native/package.json]: matches desktop now, mainline!

File: react-native/package.json [hunk 3]
```diff
@@ -9,29 +9,29 @@
...
-    "react": "15.1.0",
-    "react-native": "0.27.0",
-    "react-native-android-permissions": "0.0.3",
-    "react-native-barcodescanner": "3.0.0",
-    "react-native-camera": "git://github.com/keybase/react-native-camera.git#v0.3.9",
+    "react": "15.2.1",
+    "react-native": "0.29.2",
+    "react-native-android-permissions": "0.0.6",
+    "react-native-barcodescanner": "3.0.1",
+    "react-native-camera": "git://github.com/lwansbrough/react-native-camera.git#9ddc787cb5e40e088596dbc823583ba4aade7ecf",
```
Comment (chrisnojima) [inline react-native/package.json]: they fixed this in their repo but haven't published yet

File: shared/common-adapters/avatar.native.js [hunk 4]
```diff
@@ -27,11 +27,11 @@ export default class Avatar extends Component<void, Props, State> {
     return (
       <TouchableOpacity style={{...stylesContainer(size), ...this.props.style}} disabled={!this.props.onClick} onPress={this.props.onClick} activeOpacity={0.8}>
         <Box style={stylesContainer(size)}>
-          <Image
+          {!!uri.uri && <Image
```
Comment (chrisnojima) [inline shared/common-adapters/avatar.native.js]: if the uri isn't set  you'd error out

File: shared/common-adapters/tab-bar.native.js [hunk 5]
```diff
@@ -126,8 +126,6 @@ const stylesTabBarButtonIcon = {
   flex: 1,
   justifyContent: 'center',
   alignItems: 'center',
-  paddingRight: 22,
```
Comment (chrisnojima) [inline shared/common-adapters/tab-bar.native.js]: dunno why we have this. makes the icons not fit in the main nav

File: shared/engine/transport.native.js [hunk 6]
```diff
@@ -19,10 +20,15 @@ class MobileTransport extends BaseTransport {
   close () { }
   get_generation () { return 1 } // eslint-disable-line camelcase
 
-  _raw_write_bufs (len, buf) { // eslint-disable-line camelcase
-    const buffer = Buffer.concat([new Buffer(len), new Buffer(buf)])
-    const data = buffer.toString('base64')
-    this.writeCallback(data)
+  // We get called 2 times per transport. once with the lenth and once with the payload
```
Comment (chrisnojima) [inline shared/engine/transport.native.js]: see https://github.com/maxtaco/node-framed-msgpack-rpc/blob/master/src/packetizer.iced#L67

File: shared/globals.native.js [hunk 7]
```diff
@@ -12,3 +12,6 @@
 if (typeof __SCREENSHOT__ === 'undefined') {
   __SCREENSHOT__ = false
 }
+
+// Needed for purepack
+window.Buffer = require('buffer').Buffer
```
Comment (chrisnojima) [inline shared/globals.native.js]: workaround some node libs assuming Buffer is global, RN packager is kinda weaksauce

**Comment by chrisnojima** [comment]
- [ ] fixup android

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

File: react-native/android/app/src/main/AndroidManifest.xml [hunk 8]
```diff
@@ -7,6 +7,7 @@
 
 
     <application
+      android:name=".MainApplication"
```
Comment (chrisnojima) [inline react-native/android/app/src/main/AndroidManifest.xml]: RN android split up the MainActivity slightly: see https://github.com/facebook/react-native/releases/tag/v0.29.0

**Comment by MarcoPolo** [comment]
👍

**Comment by chrisnojima** [comment]
@MarcoPolo updated to 0.30. no changes it just worked on top of the 0.29.2 stuff

**Comment by chrisnojima** [comment]
@MarcoPolo bump on 0.30

**Comment by MarcoPolo** [comment]
👍

**Pull request merged**